File: javascript/examples/web/metaparticle-package/docker-runner.js
(function () {
    var shell = require('shelljs');

    portString = (options) => {
        if (!options || !options.ports) {
            return '';
        }
        var portArr = options.ports.map((port) => {
            return `-p ${port}:${port}`
        });
        return portArr.join(' ');
    };

    module.exports.run = (img, name, options) => {
        ports = portString(options);
        
        shell.exec(`docker run --name ${name} ${ports} -d ${img}`);
        shell.exec(`docker logs -f ${name}`, () => {
            console.log('done');
        });
    };

    module.exports.cancel = (name) => {
        shell.exec(`docker kill ${name}`);
        shell.exec(`docker rm ${name}`);
    };
})();

File: python/docker_runner.py
import os


def ports(options):
    try:
        ports_list = [str(i) for i in options['ports']]
        return ''.join(ports_list, ' ')
    except KeyError:
        return ''


def run(img, name, options):
    # Launch docker container
    os.system('docker run --rm --name {} {} -d {}'.format(
        name,
        ports(options),
        img
    ))
    # Attach to logs
    os.system('docker logs -f {}'.format(name))


def cancel(name):
    os.system('docker kill {}'.format(name))
    os.system('docker rm {}'.format(name))
